### PR TITLE
docs: update Tutorial 2

### PR DESCRIPTION
**Pull request by skrueger-ftc**
A couple of mild tweaks I felt would be beneficial after going through Tutorial 2: The Surface.

### CODE REVIEW

File: docs/beginner/tutorial2-surface/README.md [hunk 1]
```diff
@@ -223,13 +223,35 @@ Now that we've configured our surface properly we can add these new fields at th
 
 Since our `State::new()` method is async we need to change `run()` to be async as well so that we can await it.
 
+Our `window` has beened moved to the State instance, we will need to update our `event_loop` to reflect this.
+
 ```rust
 pub async fn run() {
```
Comment (sotrh) [inline docs/beginner/tutorial2-surface/README.md:229]: Something I missed was that this line needs `#[cfg_attr(target_arch = "wasm32", wasm_bindgen(start))]` before it.

File: docs/beginner/tutorial2-surface/README.md [hunk 2]
```diff
@@ -255,29 +277,11 @@ Don't use `block_on` inside of an async function if you plan to support WASM. Fu
 
 </div>
 
-If we try to build WASM now it will fail because `wasm-bindgen` doesn't support using async functions as `start` methods. You could switch to calling `run` manually in javascript, but for simplicity, we'll add the [wasm-bindgen-futures](https://docs.rs/wasm-bindgen-futures) crate to our WASM dependencies as that doesn't require us to change any code. Your dependencies should look something like this:
+<div class="note">
```
Comment (sotrh) [inline docs/beginner/tutorial2-surface/README.md:280]: Given that I need to annotate `run()` with `wasm-bindgen(start)`, this should probably just be part of the main text.

**Review by sotrh — CHANGES_REQUESTED** [review]
Just a couple notes

File: docs/beginner/tutorial2-surface/README.md (hunk 1 above)
Comment (skrueger-ftc) [reply docs/beginner/tutorial2-surface/README.md:230]: Added in 80656c7

File: docs/beginner/tutorial2-surface/README.md (hunk 2 above)
Comment (skrueger-ftc) [reply docs/beginner/tutorial2-surface/README.md:281]: I was a bit on sure on this one., as the `wasm-bindgen(start)` annotation is added in [Tutorial 1](https://github.com/sotrh/learn-wgpu/blob/a1f5995f498e5bd5cdf111155c16a681e5834396/docs/beginner/tutorial1-window/README.md?plain=1#L145). Would you prefer to drop the note tag instead?

**Comment by sotrh** [comment]
I've updated to 0.17 so you'll need to resolve some merge conflicts.

**Comment by skrueger-ftc** [comment]
I've rebased and fixed the conflict

File: docs/beginner/tutorial2-surface/README.md [hunk 3]
```diff
@@ -205,7 +207,7 @@ Regardless, `PresentMode::Fifo` will always be supported, and `PresentMode::Auto
 Now that we've configured our surface properly we can add these new fields at the end of the method.
 
 ```rust
-    async fn new(window: Window) -> Self {
+    async fn new(window: &Window) -> Self {
```
Comment (sotrh) [inline docs/beginner/tutorial2-surface/README.md:210]: State now owns `Window` so you'll need to remove the `&` here

File: docs/beginner/tutorial2-surface/README.md [hunk 4]
```diff
@@ -236,7 +261,7 @@ Now that `run()` is async, `main()` will need some way to await the future. We c
 ```toml
 [dependencies]
 # other deps...
-pollster = "0.3"
+pollster = "0.2"
```
Comment (sotrh) [inline docs/beginner/tutorial2-surface/README.md:264]: Use the later version of pollster here

**Review by sotrh — CHANGES_REQUESTED** [review]

**Review by sotrh — APPROVED** [review]

**Pull request merged**